src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2253,6 +2253,104 @@ public function get_test_authorization_header() {
 		return $result;
 	}
 
+	/**
+	 * Tests persistent_object_cache health check.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return array The health check result suggesting persistent object caching.
+	 */
+	public function get_test_health_persistent_object_cache() {
+		/**
+		 * Filters the action URL for the persistent object cache health check.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param string $action_url Learn more link for persistent object cache health check.
+		 */
+		$action_url = apply_filters(
+			'site_status_persistent_object_cache_url',
+			/* translators: Localized Support reference. */
+			__( 'https://wordpress.org/support/article/optimization/#object-caching' )
+		);
+
+		$result = array(
+			'test'        => 'persistent_object_cache',
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Performance' ),
+				'color' => 'blue',
+			),
+			'label'       => __( 'A persistent object cache is being used' ),
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( "A persistent object cache makes your site's database more efficient, resulting in faster load times because WordPress can retrieve your site's content and settings much more quickly." )
+			),
+			'actions'     => sprintf(
+				'<p><a href="%s" target="_blank" rel="noopener">%s <span class="screen-reader-text">%s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				esc_url( $action_url ),
+				__( 'Learn more about persistent object caching.' ),
+				/* translators: Accessibility text. */
+				__( '(opens in a new tab)' )
+			),
+		);
+
+		if ( wp_using_ext_object_cache() ) {
+			return $result;
+		}
+
+		if ( ! $this->should_suggest_persistent_object_cache() ) {
+			$result['label'] = __( 'A persistent object cache is not required' );
+
+			return $result;
+		}
+
+		$available_services = $this->available_object_cache_services();
+
+		$notes = __( 'Your hosting provider can tell you if persistent object caches are able to be enabled on your site.' );
+
+		if ( ! empty( $available_services ) ) {
+			$notes .= ' ' . sprintf(
+				/* translators: Available object caching services. */
+				__( 'Your host appears to support the following object caching services: %s.' ),
+				implode( ', ', $available_services )
+			);
+		}
+
+		/**
+		 * Filters the second paragraph of the health check's description
+		 * when suggesting the use of a persistent object cache.
+		 *
+		 * Hosts may want to replace the notes to recommend their preferred object caching solution.
+		 *
+		 * Plugin authors may want to append notes (not replace) on why object caching is recommended for their plugin.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param string $notes              The notes appended to the health check description.
+		 * @param array  $available_services The list of available persistent object cache services.
+		 */
+		$notes = apply_filters( 'site_status_persistent_object_cache_notes', $notes, $available_services );
```

Comment:
This filter may not be needed, since all tests can have each individual field filtered either via PHP, or JavaScript, with the `site_status_test_result` filter (which is ran on each individual test).

Unless there's a specific need for a unique one for this test specifically? If that's the case, perhaps it may even be better to expand on the existing filter to also have one that includes the test name in it for more targeted filters, such as for example `site_status_{$test_name}_test_result` (in this case becoming `site_status_persistent_object_cache_test_result`) 🤔

Reply:
Hello, @Clorith Thanks for chiming in. Good point here. I'm gonna try to explain the reason why behind it. 
We are porting these modules from the [performance plugin](https://github.com/WordPress/performance/), on there, we decided to have as many filters as needed to allow customization to this level. As we were aiming for hosting especially to active use these filters to give targeted recommendations to their clients.

Reply:
@Clorith: This filter specifically was requested by @spacedmonkey or @felixarntz IIRC to allow hosting companies to easily add their own notes to this heath check about their object caching solution.

Reply:
Makes sense in general, I'm just curious why only this test would need that special filter, is this something that should be generalized for all tests instead? And is there any reasoning for not using the existing filters (like what shortcomings were encountered, these may be shortcomings that exist for the other existing tests that we need to consider as well elsewhere).

Reply:
The `site_status_persistent_object_cache_notes` is specifically for hosting providers wanting to _only_ alter the health check `description` if the test is suggestion the use of an object cache. Using standard filters would make this quite verbose.

Reply:
+1 on keeping this filter to make it obvious and straightforward for e.g. hosting providers that would like to alter the notes. The general filter would technically work as you're outlining @Clorith, however I agree that this would be overly verbose. The notes in this Site Health check are explicitly intended to be filtered, similarly to how e.g. the PHP upgrade notice that was separately implemented a few years back allows its additional explanation to be filtered. When we eventually document the ability to filter the notes, it would be unnecessarily complex and confusing to have to explain how you have to look for the specific field to modify. In addition, by having this specific filter, we maintain the same security standard as for the default content, as the content will go through `wp_kses()` _after_ filtering. If developers used the general filter, they would have to take care of it yourself, essentially duplicating what core already does here - or worse (and maybe even more likely), they don't know and don't even use `wp_kses()`.